src/Microsoft.OData.Core/UriParser/Resolver/NormalizedModelElementsCache.cs

```diff
@@ -89,26 +93,76 @@ public List<IEdmTerm> FindTerms(string qualifiedName)
             return null;
         }
 
+        /// <summary>
+        /// Find all navigation sources that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching navigation sources, or null if no navigation source matches the name.</returns>
+        public List<IEdmNavigationSource> FindNavigationSources(string name)
+        {
+            if (navigationSourcesCache.TryGetValue(name, out List<IEdmNavigationSource> results))
+            {
+                return results;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Find all operation imports that match the <paramref name="name"/>.
+        /// </summary>
+        /// <param name="name">The case-insensitive name to match.</param>
+        /// <returns>A list of matching operation imports, or null if no operation import matches the name.</returns>
+        public List<IEdmOperationImport> FindOperationImports(string name)
+        {
+            if (operationImportsCache.TryGetValue(name, out List<IEdmOperationImport> results))
+            {
+                return results;
+            }
+
+            return null;
+        }
+
         private void PopulateSchemaElements(IEdmModel model)
         {
             foreach (IEdmSchemaElement element in model.SchemaElements)

```

Comment:
can the model be null here or the check is done by the caller?

Reply:
That is the responsibility of the caller. In this class I only placed a `Debug.Assert(model != null)` in the constructor.

That said, I checked the public methods in the public `ODataUriResolver` class and they're not performing any null-checks on the args.

Reply:
I have added null argument checks to the public methods of ODataUriResolver.